### PR TITLE
WIP: Add an emoji picker

### PR DESCRIPTION
**Pull request by agzuniverse**
This is a WIP PR that aims to create an Emoji picker that can be used to add reactions to any message.

The commit messages and the order of commits themselves are kind of messed up (After I'm entirely done I'll create a new branch and `cherry-pick` the commits in the order I want them and expand the commit messages.)

There was already a file called `EmojiPicker.js` that seemed to do nothing, so I decided make that into my component. I've made the component come up when the app opens instead of the `HomeTab` temporarily.

Right now I need to solve two problems:

* The emoji reactions are added using `emojiReactionAdd()` which needs parameters `reactionType`, `code` and `name` for an emoji. The `name` property is obtained as the `item` in the `FlatList`, but I need to obtain the `reactionType` and `code` attributes for a particular emoji as well. (I've provided an example function call)

* I need to navigate to this component on pressing the "Add Reaction" option in the action sheet that appears on long pressing a message while passing the message id so I can use it in the `emojiReactionAdd()` function call.

Can anyone give pointers on how to go about these problems?

### CODE REVIEW

File: src/i18n/translations/messages_en.json [hunk 1]
```diff
@@ -136,5 +136,6 @@
   "Do not mark messages read on scroll": "Do not mark messages read on scroll",
   "Topics": "Topics",
   "Welcome to Zulip!\n\nYou'll need to first create an account from your computer. If you're not sure where to start, go to zulipchat.com from your web browser.\n\nHope to see you back here soon!\n": "Welcome to Zulip!\n\nYou'll need to first create an account from your computer. If you're not sure where to start, go to zulipchat.com from your web browser.\n\nHope to see you back here soon!\n",
-  "Add subscribers": "Add subscribers"
+  "Add subscribers": "Add subscribers",
+  "Add Reaction": "Add Reaction"
```
Comment (borisyankov) [inline src/i18n/translations/messages_en.json]: Use a lower-case 'r' for 'reactions'.

File: src/nav/navActions.js [hunk 2]
```diff
@@ -23,6 +23,9 @@ export const navigateToSearch = (): NavigateAction =>
 export const navigateToSettings = (): NavigateAction =>
   NavigationActions.navigate({ routeName: 'settings' });
 
+export const navigateToEmojiPicker = (): NavigateAction =>
+  NavigationActions.navigate({ routeName: 'emojipicker' });
```
Comment (borisyankov) [inline src/nav/navActions.js]: Use `emoj-picker` for the name, consistent with the other naming.

File: src/main/MainTabs.js [hunk 3]
```diff
@@ -11,11 +11,12 @@ import PmConversationsCard from '../pm-conversations/PmConversationsCard';
 import SettingsCard from '../settings/SettingsCard';
 import { IconHome, IconStream, IconSettings } from '../common/Icons';
 import IconUnreadConversations from '../nav/IconUnreadConversations';
+import EmojiPicker from '../emoji/EmojiPicker';
 
 export default TabNavigator(
   {
     home: {
-      screen: HomeTab,
+      screen: EmojiPicker,
```
Comment (borisyankov) [inline src/main/MainTabs.js]: Try to do these types of 'debug convenience' changes in a separate commit, and just omit that commit from the PR. Makes it easier to review and makes the PR look closer to being done 👍

**Comment by agzuniverse** [comment]
This PR is kind of a mess, I'll `cherry-pick` the commits to a new branch and open a new PR, fixing the issues you mentioned along the way.

**Comment by jainkuniya** [comment]
> This PR is kind of a mess, I'll cherry-pick the commits to a new branch and open a new PR, fixing the issues you mentioned along the way.

@agzuniverse you can push forcefully (`-f`) to overwrite remote branch. https://stackoverflow.com/a/24768381/5612089

Creating new PR is not a better approach as it creates lots of redirection. 
Working in a same PR maintains whole history at the same place, which is helpful in the future.

**Comment by agzuniverse** [comment]
Okay I'll try to do that, in this case it was just easier to cherry pick to a different branch and by the time I read this I had already done that 😅 
Opened PR #2775

**Pull request closed without merging**